New patch (#80)
redhat-cop/operator-utils: README.md

This library covers three main areas:

1. [Utility Method](#Utility-Methods) Utility methods that are callable by any operator.
2. [Idempotent methods](#Idempotent-Methods-to-Manipulate-Resources) to manipulate resources and arrays of resources
3. [Basic operator lifecycle](#Basic-Operator-Lifecycle-Management) needs (validation, initialization, status and error management, finalization)
4. [Enforcing resources operator support](#Enforcing-Resource-Operator-Support). For those operators which calculate a set of resources that need to exist and then enforce them, generalized support for the enforcing phase is provided.

## Utility Methods

> Prior to version v2.x the general philosophy of this library was that new operator would inherint from REcorncileBase and in doing so they would havbe access to a bunch of utility methods.

Reviewer: ```suggestion
Prior to version v2.x the general philosophy of this library was that new operator would inherent from ReconcileBase and in doing so they would have access to a bunch of utility methods.
```

Author: can you fix from inherent to inherit?

Author: fixed

> With release v2.0.0 a new approach is avaibale. Utility methods are callable by any operator havintg to inehrit. This makes it easier to use this library and does not conflict with autogenerate code from kube-builder and operator-sdk.

Reviewer: ```suggestion
With release v2.0.0, a new approach is available. Utility methods are callable by any operator having to inherit. This makes it easier to use this library and does not conflict with autogenerate code from `kube-builder` and `operator-sdk`.
```

Author: can you change from `having` to `without having`?

Author: fixed

> Most of the Utility methods receive a context.Context parameter. Normally this context must be initialized with a `logr.Logger` and a `rest.Config`. Some utility methods may require more, see each individual documentation.

Reviewer: ```suggestion
Most of the Utility methods receive a `context.Context` parameter. Normally this context must be initialized with a `logr.Logger` and a `rest.Config`. Some utility methods may require more, see each individual documentation.
```

> Utility methods are current organized in the following folders:

Reviewer: ```suggestion
Utility methods are currently organized in the following folders:
```

1. crud: idempotent create/update/delete functions.

> 2. discoveryclient: methods related to the discovery client, topically used to load `apiResource` objects.

Reviewer: ```suggestion
2. discoveryclient: methods related to the discovery client, typically used to load `apiResource` objects.
```
